backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs

```diff
@@ -1,23 +1,119 @@
+                                    IsEnabled = y.IsEnabled,
+                                    Author =
+                                        new UserDto
+                                        {
+                                            UserName = y.Author.Name,
+                                            Image = y.Author.ImagePath,
+                                        },
+                                    Location = new LocationDto
+                                    {
+                                        Name = y.Location.Name,
+                                    },
+                                    Members = y.Members
+                                        .Select(m =>
+                                         new UserDto
+                                         {
+                                             UserName = m.User.Name,
+                                             Image = m.User.ImagePath,
+                                         })
+                                        .ToList()
+                                })
+                            .ToList()
+                    })
+                .ToListAsync();
+
+            return teamsWithSlots;
+        }
+
+        public async Task<ICollection<AvailabilitySlotDto>> GetAllUserAvailabilitySlotsAsync(int id)
+        {
+            var userSlots = await _context.Users
+                 .Where(x => x.Id == id)
+                 .Include(x => x.CreatedSlots)
+                    .ThenInclude(x => x.Members)
+                    .ThenInclude(x => x.User)
+                 .Include(x => x.CreatedSlots)
+                    .ThenInclude(x => x.Location)
+                 .Where(x => x.CreatedSlots.Any(x => x.Type == SlotType.Personal))
+                 .SelectMany(x => x.CreatedSlots)
+                 .Select(y =>
+                    new AvailabilitySlotDto
+                    {
+                        Id = y.Id,
+                        Name = y.Name,
+                        Type = y.Type,
+                        Size = y.Size,
+                        IsEnabled = y.IsEnabled,
+                        Author =
+                            new UserDto
+                            {
+                                UserName = y.Author.Name,
+                                Image = y.Author.ImagePath,
+                            },
+                        Location = 
+                            new LocationDto
+                            {
+                                Name = y.Location.Name,
+                            },
+                        Members = y.Members
+                            .Select(m =>
+                                new UserDto
+                                {
+                                    UserName = m.User.Name,
+                                    Image = m.User.ImagePath,
+                                })
+                            .ToList()
+                    })
```

Comment:
Could you simplify this to one layer DTO?

Reply:
Yes.
And leave projection here or use mapper?

Reply:
I simplified dto and use projection in this method, so that unused data will not be load in this method